pkg/pillar/types/newlogtypes.go

```diff
@@ -85,21 +86,43 @@ type NewlogMetrics struct {
 	AppMetrics logfileMetrics // App metrics
 }
 
-// GetTimestampFromGzipName - get timestamp from gzip file name
-func GetTimestampFromGzipName(fName string) (time.Time, error) {
-	// here are example file names:
-	// app.6656f860-7563-4bbf-8bba-051f5942982b.log.1730464687367.gz
-	// dev.log.keep.1730404601953.gz
-	// dev.log.upload.1730404601953.gz
-	// the timestamp is the number between the last two dots
-	nameParts := strings.Split(fName, ".")
-	if len(nameParts) < 2 {
-		return time.Time{}, fmt.Errorf("getTimestampFromGzipName: invalid log file name %s", fName)
+// GetTimestampFromFileName extracts a microsecond timestamp from a filename
+func GetTimestampFromFileName(filename string) (time.Time, error) {
+	// Regular expression to match a microsecond timestamp (sequence of 13-16 digits)
+	timestampRegex := regexp.MustCompile(`\b\d{13,16}\b`)
```

Comment:
Isn't it better to declare this as a variable in the file so that it is compiled at init time and not for each Get call?

Reply:
that absolutely makes sense!